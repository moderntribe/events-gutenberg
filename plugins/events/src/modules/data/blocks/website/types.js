/**
 * Internal dependencies
 */
import { PREFIX_EVENTS_STORE } from '@moderntribe/events/data/utils';

export const SET_WEBSITE_URL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_URL`;

export const SET_WEBSITE_LABEL = `${ PREFIX_EVENTS_STORE }/SET_WEBSITE_LABEL`;
