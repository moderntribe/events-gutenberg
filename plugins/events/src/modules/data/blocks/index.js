/**
 * Internal dependencies
 */
import reducer from './reducers';

export default reducer;
