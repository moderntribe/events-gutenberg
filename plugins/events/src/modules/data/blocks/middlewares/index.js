export * from './dates';
export { default as multiDay } from './multiday';
export { default as allDay } from './allday';
export * from './time';
