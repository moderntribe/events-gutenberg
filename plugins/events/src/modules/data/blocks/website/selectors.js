/**
 * External dependencies
 */
import { createSelector } from 'reselect';

export const getWebsiteBlock = ( state ) => state.blocks.website;

export const getUrl = createSelector(
	[ getWebsiteBlock ],
	( website ) => website.url,
);

export const getLabel = createSelector(
	[ getWebsiteBlock ],
	( website ) => website.label,
);
