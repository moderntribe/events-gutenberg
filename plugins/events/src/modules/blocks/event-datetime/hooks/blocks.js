/**
 * External dependencies
 */
import React from 'react';
import PropTypes from 'prop-types';

/**
 * Internal Dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';
import { PluginBlockHooks } from '@moderntribe/common/components';

const PLUGIN_TEMPLATES = {
	[ constants.EVENTS_PRO_PLUGIN ]: [
		[ 'tribe/event-pro-recurring', {}],
	],
};

const PluginDateTimeBlockHooks = () => {
	return (
		<PluginBlockHooks
			pluginTemplates={ PLUGIN_TEMPLATES }
			templateLock="all"
		/>
	);
};

PluginDateTimeBlockHooks.propTypes = {};

export default PluginDateTimeBlockHooks;
