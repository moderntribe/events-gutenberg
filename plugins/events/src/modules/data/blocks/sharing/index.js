/**
 * Internal dependencies
 */
import reducer from './reducers';

import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';

export default reducer;
export { selectors, actions, types };
