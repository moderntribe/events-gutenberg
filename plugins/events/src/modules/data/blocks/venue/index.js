/**
 * Internal dependencies
 */
import reducer from './reducers';
import * as selectors from './selectors';
import * as actions from './actions';
import * as types from './types';
import * as utils from './utils';

export default reducer;
export { selectors, actions, types, utils };
