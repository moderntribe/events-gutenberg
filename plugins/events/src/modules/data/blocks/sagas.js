
/**
 * External dependencies
 */
import { store } from '@moderntribe/common/store';

/**
 * Internal dependencies
 */
import * as price from '@moderntribe/events/data/blocks/price';
import * as venue from '@moderntribe/events/data/blocks/venue';

[
	price.sagas,
	venue.sagas,
].forEach( sagas => store.run( sagas ) );
