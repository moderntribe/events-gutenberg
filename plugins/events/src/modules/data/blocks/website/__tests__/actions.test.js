/**
 * External dependencies
 */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

/**
 * Internal dependencies
 */
import { actions } from '@@tribe/events/data/blocks/website';

const middlewares = [ thunk ];
const mockStore = configureStore( middlewares );

describe( '[STORE] - Website actions', () => {
	it( 'Should set the website URL', () => {
		expect( actions.setWebsite( 'https://tri.be/' ) ).toMatchSnapshot();
	} );

	it( 'Should set the website label', () => {
		expect( actions.setLabel( 'Modern Tribe' ) ).toMatchSnapshot();
	} );
} );

describe( '[STORE] - Website thunk actions', () => {
	it( 'Should set the initial state', () => {
		const store = mockStore( {} );

		const get = jest.fn( ( key ) => {
			const values = {
				url: 'https://tri.be/',
				urlLabel: 'Modern Tribe',
			};
			return values[ key ];
		} );

		store.dispatch( actions.setInitialState( { get } ) );

		expect( get ).toHaveBeenCalled();
		expect( get ).toHaveBeenCalledTimes( 2 );
		expect( store.getActions() ).toMatchSnapshot();
	} );
} );
