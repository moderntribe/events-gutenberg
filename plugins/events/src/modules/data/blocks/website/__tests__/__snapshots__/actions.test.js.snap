// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[STORE] - Website actions Should set the website URL 1`] = `
Object {
  "payload": Object {
    "url": "https://tri.be/",
  },
  "type": "@@MT/EVENTS/SET_WEBSITE_URL",
}
`;

exports[`[STORE] - Website actions Should set the website label 1`] = `
Object {
  "payload": Object {
    "label": "Modern Tribe",
  },
  "type": "@@MT/EVENTS/SET_WEBSITE_LABEL",
}
`;

exports[`[STORE] - Website thunk actions Should set the initial state 1`] = `
Array [
  Object {
    "payload": Object {
      "url": "https://tri.be/",
    },
    "type": "@@MT/EVENTS/SET_WEBSITE_URL",
  },
  Object {
    "payload": Object {
      "label": "Modern Tribe",
    },
    "type": "@@MT/EVENTS/SET_WEBSITE_LABEL",
  },
]
`;
